### ISSUE TITLE
azure-iothub 1.8.1 does not compile with typescript

### ISSUE THREAD

**Issue by StefanSchoof** [Azure/azure-iot-sdk-node#409]
# Context
- **OS and version used:** Ubuntu 18.04
- **Node.js version:** v10.12.0
- **npm version:** 6.4.1
- **list of installed packages:** see below 
- **cloned repo:** N/A 

# Description of the issue:
The azure-iothub package has a similar problem as in #403 .

# Code sample exhibiting the issue:
Create a index.ts in an empty folder with the content:

```
import { Client } from 'azure-iothub';
```

run

1. `npm install azure-iothub@1.7 typescript @types/node`
2. `npx tsc --init`
3. `npx tsc` (runs succesfully)
4. `npm install azure-iothub@1.8.1`
5. `npx tsc`

```
node_modules/azure-iothub/lib/amqp.d.ts:41:5 - error TS2416: Property 'connect' in type 'Amqp' is not assignable to the same property in base type 'Transport'.  Type '(done?: TripleValueCallback<Connected, IncomingMessage> | undefined) => void | Promise<ResultWithIncomingMessage<Connected>>' is not assignable to type '(done?: Callback<Connected> | undefined) => void'.
    Types of parameters 'done' and 'done' are incompatible.
      Types of parameters 'err' and 'error' are incompatible.
        Type 'Error | undefined' is not assignable to type 'Error'.
          Type 'undefined' is not assignable to type 'Error'.

41     connect(done?: TripleValueCallback<results.Connected, IncomingMessage>): Promise<ResultWithIncomingMessage<results.Connected>> | void;
       ~~~~~~~

node_modules/azure-iothub/lib/amqp.d.ts:49:5 - error TS2416: Property 'disconnect' in type 'Amqp' is not assignable to the same property in base type 'Transport'.
  Type '(done?: DoubleValueCallback<Error, Disconnected> | undefined) => void | Promise<Disconnected>' is not assignable to type '(done: Callback<Disconnected>) => void'.
    Types of parameters 'done' and 'done' are incompatible.
      Type 'Callback<Disconnected>' is not assignable to type 'DoubleValueCallback<Error, Disconnected>'.
        Types of parameters 'err' and 'result1' are incompatible.
          Type 'Error | undefined' is not assignable to type 'Error'.
            Type 'undefined' is not assignable to type 'Error'.

49     disconnect(done?: Callback<results.Disconnected>): Promise<results.Disconnected> | void;
       ~~~~~~~~~~

node_modules/azure-iothub/lib/amqp.d.ts:60:5 - error TS2416: Property 'send' in type 'Amqp' is not assignable to the same property in base type 'Transport'.
  Type '(deviceId: string, message: Message, done?: TripleValueCallback<MessageEnqueued, IncomingMessage> | undefined) => void | Promise<ResultWithIncomingMessage<MessageEnqueued>>' is not assignable to type '(deviceId: string, message: Message, done?: Callback<MessageEnqueued> | undefined) => void'.
    Types of parameters 'done' and 'done' are incompatible.
      Types of parameters 'err' and 'error' are incompatible.
        Type 'Error | undefined' is not assignable to type 'Error'.
          Type 'undefined' is not assignable to type 'Error'.

60     send(deviceId: string, message: Message, done?: IncomingMessageCallback<results.MessageEnqueued>): Promise<ResultWithIncomingMessage<results.MessageEnqueued>> | void;
       ~~~~

node_modules/azure-iothub/lib/amqp.d.ts:68:5 - error TS2416: Property 'getFeedbackReceiver' in type 'Amqp' is not assignable to the same property in base type 'Transport'.
  Type '(done?: TripleValueCallback<ServiceReceiver, IncomingMessage> | undefined) => void | Promise<ResultWithIncomingMessage<ServiceReceiver>>' is not assignable to type '(done: Callback<ServiceReceiver>) => void'.
    Types of parameters 'done' and 'done' are incompatible.
      Type 'Callback<ServiceReceiver>' is not assignable to type 'TripleValueCallback<ServiceReceiver, IncomingMessage>'.
        Types of parameters 'err' and 'error' are incompatible.
          Type 'Error | undefined' is not assignable to type 'Error'.
            Type 'undefined' is not assignable to type 'Error'.

68     getFeedbackReceiver(done?: IncomingMessageCallback<Client.ServiceReceiver>): Promise<ResultWithIncomingMessage<Client.ServiceReceiver>> | void;
       ~~~~~~~~~~~~~~~~~~~

node_modules/azure-iothub/lib/amqp.d.ts:76:5 - error TS2416: Property 'getFileNotificationReceiver' in type 'Amqp' is not assignable to the same property in base type 'Transport'.
  Type '(done?: TripleValueCallback<ServiceReceiver, IncomingMessage> | undefined) => void | Promise<ResultWithIncomingMessage<ServiceReceiver>>' is not assignable to type '(done: Callback<ServiceReceiver>) => void'.
    Types of parameters 'done' and 'done' are incompatible.
      Type 'Callback<ServiceReceiver>' is not assignable to type 'TripleValueCallback<ServiceReceiver, IncomingMessage>'.

76     getFileNotificationReceiver(done?: IncomingMessageCallback<Client.ServiceReceiver>): Promise<ResultWithIncomingMessage<Client.ServiceReceiver>> | void;
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/azure-iothub/lib/amqp_ws.d.ts:7:22 - error TS2420: Class 'AmqpWs' incorrectly implements interface 'Transport'.
  Types of property 'connect' are incompatible.
    Type '(done?: TripleValueCallback<Connected, IncomingMessage> | undefined) => void | Promise<ResultWithIncomingMessage<Connected>>' is not assignable to type '(done?: Callback<Connected> | undefined) => void'.
      Types of parameters 'done' and 'done' are incompatible.
        Types of parameters 'err' and 'error' are incompatible.
          Type 'Error | undefined' is not assignable to type 'Error'.
            Type 'undefined' is not assignable to type 'Error'.

7 export declare class AmqpWs extends Amqp implements Client.Transport {
                       ~~~~~~
```

# Console log of the issue:

**Comment by anthonyvercolano** [comment]
Sigh.  We'll have something out by the end of 2nd or 3rd.

**Comment by StefanSchoof** [comment]
The old version still works, so this is for me not critical.

**Comment by anthonyvercolano** [comment]
Merged fix into master.

**Comment by pierreca** [comment]
@StefanSchoof we just released version 1.8.2 which should solve these errors. closing the issue.

**Comment by az-iot-builder-01** [comment]
@StefanSchoof, thank you for your contribution to our open-sourced project! Please help us improve by filling out this 2-minute [customer satisfaction survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbRx8jpG79OudPqY-BojcKjkJUMFFBQjgxNkw5ODJIRjBWTzI0Nk85R0pUQiQlQCN0PWcu)
